src/status_im/contexts/settings/language_and_currency/currency/view.cljs

```diff
@@ -0,0 +1,95 @@
+            [status-im.contexts.settings.language-and-currency.currency.utils :as utils]
+            [utils.i18n :as i18n]
+            [utils.navigation :as navigation]
+            [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [[search-text set-search-text] (rn/use-state "")
+        on-currency-press             (rn/use-callback #(rf/dispatch [:profile.settings/update-currency
+                                                                      %]))
+        selected-currency             (rf/sub [:profile/currency-info])
+        filtered-currencies           (rn/use-memo
+                                       #(->> currency/currencies
+                                             (filter (fn [[_ currency]]
+                                                       (let [search-lc (string/lower-case search-text)]
+                                                         (or (string/includes? (string/lower-case
+                                                                                (:display-name currency))
+                                                                               search-lc)
+                                                             (string/includes? (string/lower-case
+                                                                                (:code currency))
+                                                                               search-lc)))))
+                                             (map second))
+                                       [search-text])
+        popular-currencies            (rn/use-memo
+                                       #(filter :selected filtered-currencies)
+                                       [filtered-currencies])
+        crypto-currencies             (rn/use-memo
+                                       #(filter :is-token filtered-currencies)
+                                       [filtered-currencies])
+        all-currencies                (remove #(or (:selected %) (:is-token %)) filtered-currencies)]
+    [quo/overlay
+     {:type       :shell
+      :top-inset? true}
+     [quo/page-nav
+      {:background :blur
+       :icon-name  :i/arrow-left
+       :on-press   navigation/navigate-back}]
+     [quo/page-top
+      {:title (i18n/label :t/currency)}]
+     [quo/input
+      {:small?          true
+       :blur?           true
+       :container-style style/input-container
+       :placeholder     (i18n/label :t/search-currencies)
+       :icon-name       :i/search
+       :value           search-text
+       :on-change-text  set-search-text}]
+     (when (empty? filtered-currencies)
+       [quo/empty-state
+        {:title           (i18n/label :t/no-result)
+         :image           (resources/get-themed-image :no-assets :dark)
+         :container-style style/empty-results
+         :description     (i18n/label :t/try-with-different-currency)}])
+     [gesture/scroll-view
```

Comment:
Do we need the gesture scroll view here? 🤔

Reply:
Yes otherwise user can't scroll, if will check if i can switch to flat-list